geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStoreKeywordsTest.scala

```diff
@@ -0,0 +1,147 @@
+
+      val initialKeywords = Set("keywordA=Hello", "keywordB", "keywordC")
+      val spec = s"name:String;$Keywords=${initialKeywords.mkString(KeywordsDelimiter)}"
+      val sft = SimpleFeatureTypes.mutable(createNewSchema(spec))
+
+      val keywordsToRemove = Set("keywordA=Hello", "keywordC")
+      sft.removeKeywords(keywordsToRemove)
+      ds.updateSchema(sft.getTypeName, sft)
+
+      val keywords = ds.getFeatureSource(sft.getTypeName).getInfo.getKeywords.toSeq
+      keywords must contain(initialKeywords -- keywordsToRemove)
+      keywords must not(contain(keywordsToRemove))
+    }
+
+    "add keywords to schema" in {
+      import org.locationtech.geomesa.utils.geotools.SimpleFeatureTypes.Configs.Keywords
+
+      val originalKeyword = "keywordB"
+      val keywordsToAdd = Set("keywordA", "~!@#$%^&*()_+`=/.,<>?;:|[]{}\\")
+
+      val spec = s"name:String;$Keywords=$originalKeyword"
+      val sft = SimpleFeatureTypes.mutable(createNewSchema(spec))
+
+      sft.addKeywords(keywordsToAdd)
+      ds.updateSchema(sft.getTypeName, sft)
+
+      val fs = ds.getFeatureSource(sft.getTypeName)
+      fs.getInfo.getKeywords.toSeq must containAllOf((keywordsToAdd + originalKeyword).toSeq)
+    }
+
+    "not allow updating non-keyword user data" in {
+      import org.locationtech.geomesa.utils.geotools.SimpleFeatureTypes.Configs.TableSharing
+
+      val sft = SimpleFeatureTypes.mutable(createNewSchema("name:String"))
+      ds.createSchema(sft)
+
+      sft.getUserData.put(TableSharing, "false") // Change table sharing
+
+      ds.updateSchema(sft.getTypeName, sft) must throwAn[UnsupportedOperationException]
+    }
+
+    "create config from schema w/ keywords" in {
+      val confString = """
+                         |{
+                         |  type-name = "testconf"
+                         |  fields = [
+                         |    { name = "testStr",  type = "string"       , index = true  },
+                         |    { name = "testCard", type = "string"       , index = true, cardinality = high },
+                         |    { name = "testList", type = "List[String]" , index = false },
+                         |    { name = "geom",     type = "Point"        , srid = 4326, default = true }
+                         |  ]
+                         |  user-data = {
+                         |    geomesa.keywords = ["keywordA=foo,bar","keywordB","keywordC"]
+                         |  }
+                         |}
+         """.stripMargin
+      val regular = ConfigFactory.parseString(confString)
+      val sftWithKeywords = SimpleFeatureTypes.createType(regular)
+
+      // Currently breaks as it can't derive the type name from the config string
+      val newSft = SimpleFeatureTypes.createType(SimpleFeatureTypes.toConfig(sftWithKeywords))
+
+      sftWithKeywords.getKeywords mustEqual newSft.getKeywords
+    }.pendingUntilFixed
```

Comment:
Is this newly broken?

Either way, I'd love to see a ticket for it if we intend to fix it.

Reply:
this was just copied over from the old test case. i'm not sure what the issue is exactly? since we've de-emphasized keywords, i would say we don't need to fix it.